### ISSUE TITLE
migration fails on sqlite

### ISSUE THREAD

**Issue by josephmancuso** [MasoniteFramework/orm#172]
```python
        with self.schema.table('videos') as table:
            table.unsigned_integer('playlist_id').nullable()
            table.foreign('playlist_id').references('id').on('playlists')
```

```
ALTER TABLE "videos" ADD "playlist_id" UNSIGNED INT, ADD CONSTRAINT videos_playlist_id_foreign FOREIGN KEY ("playlist_id") REFERENCES "playlists"("id") 
```

```
near ",": syntax error
```

**Comment by Marlysson** [comment]
@josephmancuso Can I take this?
Investigating about this, I found that we should enable foreign key support in sqlite databases through  "PRAGMA foreign_keys = ON". This would be done in each opened connection.

https://sqlite.org/foreignkeys.html

**Comment by josephmancuso** [comment]
oh you know what, you are actually right. I thought it was more complex than that.

We might be able to just straight up put it in the alter table method in the grammar class

**Comment by Marlysson** [comment]
@josephmancuso The error about ```near ",": syntax error``` is that sqlite doesn't support add constraints in a table through ```ALTER TABLE```. The recommended is defined it when create the table ( with CREATE TABLE ) , but the case of alter table to add new foreign keys is so common that this suggestion don't should be considered.

What should we do?

Here is the reference: https://www.sqlite.org/omitted.html

**Comment by josephmancuso** [comment]
@Marlysson in the SQLite grammar class we just need to add that 

```python
def alter_format(self):
        return "PRAGMA foreign_keys = ON; ALTER TABLE {table} {columns}{constraints}{foreign_keys}"
```

**Comment by josephmancuso** [comment]
we might need to also put it back off at the end of the query?

**Comment by Marlysson** [comment]
Yes, the problem is when the query uses ```alter table``` together ```add constraint``` the sqlite doesn't support.
But I'll try what you suggested.

By that I read, it's just run once on connection and the sqlite database is supported with foreign keys.

**Comment by josephmancuso** [comment]
oh hmm ok

**Comment by Marlysson** [comment]
Hmm, by changing in the grammars, all tests should be changed too ( 19 more exactly until now ).

Is recommended to set that once in connection ( when connect to database maybe) ?

I think this bug is more "complex" than I've found because:

> The ALTER command in SQLite3 is limited to adding columns or renaming tables, so you cannot do this. Instead, you must re-create the table with the desired constraints.

The only operations in alter table are ( add table and columns, rename columns ), there's no way add constraint in this statement. [possible operations in ALTER TABLE ](https://www.sqlite.org/lang_altertable.html)

There's some suggestions to solve this:

1. Create a index instead of a constraint ( until I saw, there's will be same effect ).
```
CREATE UNIQUE INDEX videos_playlist_index ON videos(playlist_id);
```

2. Create a constraint to column level ( this works very fine )
```
ALTER TABLE videos ADD playlist_id UNSIGNED INT REFERENCES playlists(id);
```
Am I right in implement the option 2 instead 1?

Just more one question before implement: the field types in sqlite doesn't support ```unsigned int```, it's converted to ```integer```... change this behavior after solve this issue?

**Comment by josephmancuso** [comment]
> Just more one question before implement: the field types in sqlite doesn't support unsigned int, it's converted to integer... change this behavior after solve this issue?

Yes if the data type is not supported it can be mapped in the `type_map` on the grammar

> ALTER TABLE videos ADD playlist_id UNSIGNED INT REFERENCES playlists(id);

Yes if this is the same thing as what we are trying to do then sure

**Comment by josephmancuso** [comment]
Fixed in #181
